src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,40 @@
+import { memo, useRef } from 'react';
+import Button from '../Button/Button';
+import TopControls from './subComponents/TopControls/TopControls';
+import ControlButtons from './subComponents/ControlButtons/ControlButtons';
+import EventTitle from './subComponents/EventTitle/EventTitle';
+import EventSettings from './subComponents/EventSettings/EventSettings';
+import { useModal } from '../../hooks/useModal';
+
+import './modal.scss';
+import type { InputRefs } from '../../types/main';
```

Comment:
If you have `types/index.ts` file you don't need main file
```suggestion
import type { InputRefs } from '../../types';
```